### ISSUE TITLE
Multiple rounds of finetuning, on different datasets

### ISSUE THREAD

**Issue by david-sg** [minimaxir/gpt-2-simple#65]
I'm trying to train the model to produce tweets which resemble resemble tweets by a fairly prolific content creator.
My datasets:
Tweets: 7000 (171k words, 0.99mb)
Reddit posts: 5000 lines (336k words, 1.66mb)
Articles (each paragraph 1 line): 2100 lines (165k words, 0.97mb)

I've tried finetuning with 2000 steps on the articles, then 2000 steps on the reddit posts, and finally 6000 on the tweets (with very low learning rates due to small sample sizes). 

Any thoughts on if this is the best methodology? Does subsequent finetuning on the new datasets negate the earlier training?

Is there a suggested loss that I should aim for on the steps?

**Comment by woctezuma** [comment]
> Any thoughts on if this is the best methodology?

If the goal is to generate tweets, why not fine-tune solely on these tweets?
Or is the goal to generate tweets on a given subject, e.g. by feeding Reddit posts about this subject and then feeding tweets?

**Comment by david-sg** [comment]
I was concerned that the vocabulary and language structure unique to this person wouldn't be sufficient using tweets alone.

**Comment by PyxAI** [comment]
I am tackling a similar issue
I have a huge dataset (GB's) so i've split it into smaller files
Trying to train on individual files is unsuccessful so far, with issues that I currently weren't able to overcome.
Any idea of best practice?

**Comment by johndpope** [comment]
This aloha-chatbot repo by @newpro may have a bitter fit for above use case - 
https://github.com/newpro/aloha-chatbot
do share results if you get any mileage.
